sdk/attestation/attestation/src/models/attestationSigner.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ */
+
+import {JsonWebKey} from "../generated/models";
+import {base64DecodeString} from "../utils/base64";
+
+export class AttestationSigner
+{
+    constructor(key : JsonWebKey) {
+        if (key.kid === null) {
+            throw new Error("KeyID field in signer must not be null.")
+        }
+        else if (key.kid === undefined) {
+            throw new Error("KeyID field in signer cannot be undefined.")
+        }
+        else
+        {
+            this.key_id = key.kid.toString();
+        }
+
+        this.certificates = new Array(key.x5C?.length)
```

Comment:
Interestingly, this will create an array of length 1 if `key.x5C` is undefined. Have you considered using map?
```js
this.certificates = key.x5C?.map(base64DecodeString) ?? []
```

Reply:
Isn't a map a key/value combination? How would I implement this as a map? The x5c values are an array of base64 encoded X.509 certificates, with x5c[1] being the signer, and the remaining array entries representing an X.509 certificate chain.

If implemented as a map, I don't know what the key would be (I guess it could be an index, but that would be *weird*). And a map loses the ordering.

I'm open to suggestions though.

Reply:
In JavaScript, calling the `map` method of an array is used to transform the contents of an array. So the snippet I wrote above would essentially create a new array that is the result of executing `base64DecodeString` on each element of the `x5C` array.

Reply:
Sorry, was confused. Makes a lot of sense.